anvil/src/udev.rs

```diff
@@ -799,6 +826,86 @@ enum DeviceAddError {
     AddNode(egl::Error),
 }
 
+fn get_surface_dmabuf_feedback(
+    primary_gpu: DrmNode,
+    render_node: DrmNode,
+    gpus: &mut GpuManager<GbmGlesBackend<Gles2Renderer>>,
+    composition: &SurfaceComposition,
+) -> Option<DrmSurfaceDmabufFeedback> {
+    let primary_formats = gpus
+        .single_renderer(&primary_gpu)
+        .ok()?
+        .dmabuf_formats()
+        .copied()
+        .collect::<HashSet<_>>();
+
+    let render_formats = gpus
+        .single_renderer(&render_node)
+        .ok()?
+        .dmabuf_formats()
+        .copied()
+        .collect::<HashSet<_>>();
+
+    let all_render_formats = primary_formats
+        .iter()
+        .copied()
+        .chain(render_formats.iter().copied())
+        .collect::<HashSet<_>>();
+
+    let surface = composition.surface();
+    let planes = surface.planes().unwrap();
+    // We limit the scan-out trache to formats we can also render from
+    // so that there is always a fallback render path available in case
+    // the supplied buffer can not be scanned out directly
+    let planes_formats = surface
+        .supported_formats(planes.primary.handle)
+        .unwrap()
+        .into_iter()
+        .chain(
+            planes
+                .overlay
+                .iter()
+                .flat_map(|p| surface.supported_formats(p.handle).unwrap()),
+        )
+        .collect::<HashSet<_>>()
+        .intersection(&all_render_formats)
+        .copied()
+        .collect::<Vec<_>>();
+
+    let builder = DmabufFeedbackBuilder::new(primary_gpu.dev_id(), primary_formats);
+    let render_feedback = builder
+        .clone()
+        .add_preference_tranche(render_node.dev_id(), None, render_formats.clone())
+        .build()
+        .unwrap();
+
+    // We only provide a scan-out tranche if the drm node we selected as the preference for
+    // rendering equals the scan-out drm node
+    let can_scanout = render_node
+        .node_with_type(NodeType::Primary)
+        .and_then(|n| n.ok().map(|n| n.dev_id()))
+        == surface.device_fd().dev_id().ok();
```

Comment:
I think this whole check is unnecessary.

Given we could scan out on the secondary gpu as well - if it has outputs - and plane_formats are filtered for render-able formats anyway, we can just advertise the scanout tranche without any restrictions.

Reply:
right, this was added with optimus setup in mind. 
removed the check